0004-metric-configurable-aggregation.md

```diff
@@ -0,0 +1,59 @@
+# Let Metrics support configrable, recommended aggregations
+
+Let the user configure recommended Metric aggregations (SUM, COUNT, MIN, MAX, LAST_VALUE, HISTOGRAM, SUMMARY).
+
+## Motivation
+
+In the current API proposal, Metric types like Gauge and Cumulative are mapped into specific aggregations: Gauge:LAST_VALUE and Cumulative:SUM.  Depending on RFC 0002-metric-measure, which creates a new MeasureMetric type, this proposal introduces the ability to configure alternative, potentially multiple aggregations for Metrics.  This allows the MeasureMetric type to support HISTOGRAM and SUMMARY aggregations, as an alternative to raw statistics.
+
+## Explanation
+
+This proposal completes the elimination of Raw statistics by recognizing that aggregations should be independent of metric type.  This recognizes that _sometimes_ we have a cumulative but want to compute a histogram of increment values, and _sometimes_ we have a measure that has multiple interesting aggregations.
+
+Following this change, we should think of the _Metric type_ as:
+
+1. Indicating something about what kind of numbers are being recorded (i.e., the input domain, e.g., restricted to values >= 0?)
+  1. For Gauges: Something pre-computed where rate or count is not relevant
+  1. For Cumulatives: Something where rate or count is relevant
+  1. For Measures: Something where individual values are relevant
+1. Indicating something about the default interpretation, based on the action verb (Set, Inc, Record, etc.)
+  1. For Gauges: the action is Set()
+  1. For Cumulatives: the action is Inc()
+  1. For Measures: the action is Record()
+1. Unless the programmer declares otherwise, suggesting a default aggregation
+  1. For Gauges: LAST_VALUE is interesting, SUM is likely not interesting
+  1. For Cumulatives: SUM is interesting, LAST_VALUE is likely not interesting
+  1. For Measures: all aggregations apply, default is MIN, MAX, SUM, COUNT.
+
+## Internal details
+
+Metric constructors should take an optional list of aggregations, to override the default behavior.  When constructed with an explicit list of aggregations, the implementation may use this as a hint about which aggregations should be exported by default.  However, the implementation is not bound by these recommendations in any way and is free to control which aggregations that are applied.
```

Comment:
Might be worth considering an alternative (at least farther down the line) in which the hints are explicitly provided on export, but not necessarily used directly by OpenTelemetry.

(This suggestion is mostly a continuation of me thinking that OpenTelemetry should offload metric state management to exporters, so take it with however many grains of salt you want.)

Reply:
I think you (and I!) would both like to argue for lower-level exporters. :+1: